src/emmy/complex/impl.cljc

```diff
@@ -0,0 +1,680 @@
+   arithmetic for Emmy. The authors are indebted to Robert Eisele's
+   `Complex.js` implementation from which we have borrowed freely and
+   which is licensed as follows:
+
+   * Copyright (c) 2020, Robert Eisele (robert@xarg.org)
+   * Dual licensed under the MIT or GPL Version 2 licenses."
+  (:refer-clojure :exclude [abs zero?])
+  (:require [emmy.generic :as g]
+            [emmy.util :as u]
+            [emmy.value :as v]))
+
+(declare equal?)
+
+(deftype Complex [re im]
+  v/IKind
+  (kind [_] :emmy.complex/complex)
+
+  v/Numerical
+  (numerical? [_] true)
+
+  #?@(:clj [Object
+            (equals [a b] (equal? a b))]
+      :cljs [IEquiv
+             (-equiv [a b] (equal? a b))
+
+             IPrintWithWriter
+             (-pr-writer
+              [z writer _]
+              (write-all
+               writer
+               "#emmy/complex "
+               (str [(.-re z) (.-im z)])))]))
+
+(def ZERO (Complex. 0 0))
+(def ONE (Complex. 1 0))
+(def I (Complex. 0 1))
+(def -I (Complex. 0 -1))
+(def INFINITY (Complex. ##Inf ##Inf))
+(def NAN (Complex. ##NaN ##NaN))
+(def LN2 (Math/log 2))
+
+(def ^:private PI:2 (/ Math/PI 2))
+(def ^:private PI:4 (/ Math/PI 4))
+
+(defn equal?
+  "Test for complex equality with another object."
+  [^Complex a b]
+  (cond (instance? Complex b)
+        (and (v/= (.-re a) (.-re b))
+             (v/= (.-im a) (.-im b)))
+
+        (v/real? b)
+        (and (g/zero? (.-im a))
+             (v/= (.-re a) b))
+
+        :else (v/= a b)))
+
+(defn zero?
+  "Determines whether or not a complex number is at the zero pole of the
+  Riemann sphere."
+  [^Complex z]
+  (and (g/zero? (.-re z))
+       (g/zero? (.-im z))))
+
+(defn nan?
+  "Determines whether a complex number is not on the Riemann sphere."
+  [^Complex z]
+  (or (u/nan? (.-re z))
+      (u/nan? (.-im z))))
+
+(comment
+  (require '[emmy.series :as s])
+  (take 10 (g/- s/cos-series 1))
+  ;; => (0 0 -1/2 0 1/24 0 -1/720 0 1/40320 0)
+  ;; From this we can see that we may regard the expansion as a series in x^2, with a
+  ;; zero constant term.
+  (->> (g/- s/cos-series 1)
+       (remove g/zero?)  ;; eliminate the useless zero coefficients of the odd powers of x
+       (map double)      ;; we don't want the rational arithmetic to survive
+       (take 8)          ;; the filtered series now has coefficients for x^2, x^4 ... x^16
+       (cons 0.)         ;; cons 0 and reverse facilitate evaluation with Horner's method
+       (reverse))
+  ;; => (4.779477332387385E-14 -1.147074559772972E-11 2.08767569878681E-9 -2.755731922398589E-7
+  ;;     2.48015873015873E-5 -0.001388888888888889 0.04166666666666667 -0.5 0.0)
+  ;; We copy this sequence, in order to avoid the dependency complex -> series,
+  ;; which creates a cycle that would be difficult to break, and it's clear in
+  ;; this case that complex is lower in the abstraction hierarchy than series.
+  )
+
+(def ^:private cos-1-square-terms
```

Comment:
you might make this `:no-doc` so you can use it from other test namespaces, and then put the above into the series tests, with a note here pointing there?